plugins/woocommerce-docs/client/index.tsx

```diff
@@ -1,25 +1,110 @@
 /**
  * External dependencies
  */
-import 'react';
+import React, { useState } from 'react';
 import ReactDOM from 'react-dom';
+import {
+	Button,
+	__experimentalItem as Item,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalHeading as Heading,
+	Card,
+	CardBody,
+	CardHeader,
+	CardFooter,
+} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import { useManifests } from './data/useManifests';
+import { isURL } from './util/url';
+import { useJobLog } from './data/useJobs';
 
 const App = () => {
-	const { isLoading, manifests } = useManifests();
+	const { isLoading, manifests, createManifest, deleteManifest } =
+		useManifests();
+	const [ newManifest, setNewManifest ] = useState< string >( '' );
+	const { jobs, isLoading: jobsLoading } = useJobLog();
 
 	return (
 		<>
-			<h1>WooCommerce Docs Administration</h1>
-			<p>Manifests:</p>
-			{ ! isLoading &&
-				manifests.map( ( manifest ) => (
-					<p key={ manifest }>{ manifest }</p>
-				) ) }
+			<Heading level={ 1 }>WooCommerce Docs Administration</Heading>
+
+			{ isLoading && <p>Loading...</p> }
+
+			<Card elevation={ 3 }>
+				<CardHeader>
+					<h2>Manifests</h2>
+				</CardHeader>
+				<CardBody>
+					<ItemGroup>
+						{ ! isLoading &&
+							!! manifests.length &&
+							manifests.map( ( [ manifestUrl ] ) => (
+								<Item key={ manifestUrl }>
+									{ manifestUrl }
+									<Button
+										variant="tertiary"
+										onClick={ () => {
+											deleteManifest( manifestUrl );
+										} }
+									>
+										Remove this manifest
+									</Button>
+								</Item>
+							) ) }
+						{ ! isLoading && manifests.length === 0 && (
+							<p>No manifests found.</p>
+						) }
+					</ItemGroup>
+				</CardBody>
+				<CardFooter>
+					<input
+						type="text"
+						value={ newManifest }
+						onChange={ ( e ) => setNewManifest( e.target.value ) }
+					/>
+					{ !! newManifest.length && ! isURL( newManifest ) && (
+						<p>Invalid URL</p>
+					) }
+					<Button
+						variant="primary"
+						disabled={
+							! newManifest &&
+							! newManifest.length &&
+							! isURL( newManifest )
+						}
+						onClick={ () => {
+							createManifest( newManifest );
+							setNewManifest( '' );
+						} }
+					>
+						Add manifest URL
+					</Button>
+				</CardFooter>
+			</Card>
+
+			<Card elevation={ 3 }>
+				<CardHeader>
+					<h2>Job Log</h2>
+				</CardHeader>
+				<CardBody>
+					<ItemGroup>
+						{ ! jobsLoading &&
+							jobs.map( ( job ) => (
+								<div key={ job.action_id }>
```

Comment:
I'm getting duplicate key errors in the console and can't find a key for the job messages in React developer tools. Does the key need to go in the Item here, maybe?

Reply:
Yeah so this was something I did early on when I didn't understand that action_id was an ID tied to the recurring action which will always be the same. I've changed this to be a key that is a combo of index and the date/time of the log which gets rid of the React error you mentioned here.